--- FusionSound-1.1.1/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c.orig	2009-10-26 16:05:19.965978378 +0100
+++ FusionSound-1.1.1/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c	2009-10-26 16:10:30.944605524 +0100
@@ -123,8 +123,8 @@
      return len;
 }
 
-static offset_t
-av_seek_callback( void *opaque, offset_t offset, int whence )
+static int64_t
+av_seek_callback( void *opaque, int64_t offset, int whence )
 {
      IFusionSoundMusicProvider_FFmpeg_data *data = opaque;
      unsigned int                           pos  = 0;
