--- FusionSound-1.1.1/proxy/dispatcher/ifusionsound_dispatcher.c.orig	2010-11-18 08:56:24.443104966 +0100
+++ FusionSound-1.1.1/proxy/dispatcher/ifusionsound_dispatcher.c	2010-11-18 09:17:35.627103290 +0100
@@ -207,7 +207,7 @@ Dispatch_GetDeviceDescription( IFusionSo
      if (ret)
           return ret;
           
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     DR_OK, VOODOO_INSTANCE_NONE,
                                     VMBT_DATA, sizeof(FSDeviceDescription), &desc,
                                     VMBT_NONE );
@@ -243,7 +243,7 @@ Dispatch_CreateBuffer( IFusionSound *thi
           return ret;
      }
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     DR_OK, instance,
                                     VMBT_DATA, sizeof(FSBufferDescription), &dsc,
                                     VMBT_NONE );
@@ -279,7 +279,7 @@ Dispatch_CreateStream( IFusionSound *thi
           return ret;
      }
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     DR_OK, instance,
                                     VMBT_DATA, sizeof(FSStreamDescription), &dsc,
                                     VMBT_NONE );
@@ -312,7 +312,7 @@ Dispatch_CreateMusicProvider( IFusionSou
           return ret;
      }
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     DR_OK, instance,
                                     VMBT_NONE );
 }
@@ -328,7 +328,7 @@ Dispatch_GetMasterVolume( IFusionSound *
      
      ret = real->GetMasterVolume( real, &level );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_DATA, sizeof(float), &level, 
                                     VMBT_NONE );
@@ -350,7 +350,7 @@ Dispatch_SetMasterVolume( IFusionSound *
      
      ret = real->SetMasterVolume( real, level );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -366,7 +366,7 @@ Dispatch_GetLocalVolume( IFusionSound *t
      
      ret = real->GetLocalVolume( real, &level );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_DATA, sizeof(float), &level, 
                                     VMBT_NONE );
@@ -388,7 +388,7 @@ Dispatch_SetLocalVolume( IFusionSound *t
      
      ret = real->SetLocalVolume( real, level );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -403,7 +403,7 @@ Dispatch_Suspend( IFusionSound *thiz, IF
      
      ret = real->Suspend( real );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -418,7 +418,7 @@ Dispatch_Resume( IFusionSound *thiz, IFu
      
      ret = real->Resume( real );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
--- FusionSound-1.1.1/proxy/dispatcher/ifusionsoundbuffer_dispatcher.c.orig	2010-11-18 08:56:24.447102452 +0100
+++ FusionSound-1.1.1/proxy/dispatcher/ifusionsoundbuffer_dispatcher.c	2010-11-18 09:18:49.479101613 +0100
@@ -186,7 +186,7 @@ Dispatch_GetDescription( IFusionSoundBuf
      if (ret)
           return ret;
           
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     DR_OK, VOODOO_INSTANCE_NONE,
                                     VMBT_DATA, sizeof(FSBufferDescription), &desc,
                                     VMBT_NONE );
@@ -208,7 +208,7 @@ Dispatch_SetPosition( IFusionSoundBuffer
      
      ret = real->SetPosition( real, position );
           
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -250,7 +250,7 @@ Dispatch_Unlock( IFusionSoundBuffer *thi
      
      real->Unlock( real );
      
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     DR_OK, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -282,7 +282,7 @@ Dispatch_Unlock_DPACK( IFusionSoundBuffe
      
      real->Unlock( real );
      
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     DR_OK, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -303,7 +303,7 @@ Dispatch_Play( IFusionSoundBuffer *thiz,
      
      ret = real->Play( real, flags );
      
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -318,7 +318,7 @@ Dispatch_Stop( IFusionSoundBuffer *thiz,
      
      ret = real->Stop( real );
      
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -344,7 +344,7 @@ Dispatch_CreatePlayback( IFusionSoundBuf
           return ret;
      }
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     DR_OK, instance,
                                     VMBT_NONE );
 }
--- FusionSound-1.1.1/proxy/dispatcher/ifusionsoundplayback_dispatcher.c.orig	2010-11-18 08:56:24.447102452 +0100
+++ FusionSound-1.1.1/proxy/dispatcher/ifusionsoundplayback_dispatcher.c	2010-11-18 09:19:48.907101613 +0100
@@ -211,7 +211,7 @@ Dispatch_Start( IFusionSoundPlayback *th
      
      ret = real->Start( real, start, stop );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -226,7 +226,7 @@ Dispatch_Stop( IFusionSoundPlayback *thi
      
      ret = real->Stop( real );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -241,7 +241,7 @@ Dispatch_Continue( IFusionSoundPlayback
      
      ret = real->Continue( real );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -256,7 +256,7 @@ Dispatch_Wait( IFusionSoundPlayback *thi
      
      ret = real->Wait( real );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -275,7 +275,7 @@ Dispatch_GetStatus( IFusionSoundPlayback
      if (ret)
           return ret;
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     DR_OK, VOODOO_INSTANCE_NONE,
                                     VMBT_INT, playing,
                                     VMBT_INT, position,
@@ -298,7 +298,7 @@ Dispatch_SetVolume( IFusionSoundPlayback
      
      ret = real->SetVolume( real, level );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -319,7 +319,7 @@ Dispatch_SetPan( IFusionSoundPlayback *t
      
      ret = real->SetPan( real, value );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -340,7 +340,7 @@ Dispatch_SetPitch( IFusionSoundPlayback
      
      ret = real->SetPitch( real, value );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -361,7 +361,7 @@ Dispatch_SetDirection( IFusionSoundPlayb
      
      ret = real->SetDirection( real, direction );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -384,7 +384,7 @@ Dispatch_SetDownmixLevels( IFusionSoundP
      
      ret = real->SetDownmixLevels( real, center, rear );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
--- FusionSound-1.1.1/proxy/dispatcher/ifusionsoundstream_dispatcher.c.orig	2010-11-18 08:56:24.475097423 +0100
+++ FusionSound-1.1.1/proxy/dispatcher/ifusionsoundstream_dispatcher.c	2010-11-18 09:20:43.463101613 +0100
@@ -216,7 +216,7 @@ Dispatch_GetDescription( IFusionSoundStr
      if (ret)
           return ret;
           
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     DR_OK, VOODOO_INSTANCE_NONE,
                                     VMBT_DATA, sizeof(FSStreamDescription), &desc,
                                     VMBT_NONE );
@@ -240,7 +240,7 @@ Dispatch_Write( IFusionSoundStream *thiz
      
      ret = real->Write( real, samples, length );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -268,7 +268,7 @@ Dispatch_Write_DPACK( IFusionSoundStream
      
      ret = real->Write( real, samples, length );
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -289,7 +289,7 @@ Dispatch_Wait( IFusionSoundStream *thiz,
      
      ret = real->Wait( real, length );
      
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -309,7 +309,7 @@ Dispatch_GetStatus( IFusionSoundStream *
 
      ret = real->GetStatus( real, &filled, &total, &read_pos, &write_pos, &playing );
      
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_INT, filled,
                                     VMBT_INT, total,
@@ -329,7 +329,7 @@ Dispatch_Flush( IFusionSoundStream *thiz
      
      ret = real->Flush( real );
      
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -344,7 +344,7 @@ Dispatch_Drop( IFusionSoundStream *thiz,
      
      ret = real->Drop( real );
      
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_NONE );
 }
@@ -360,7 +360,7 @@ Dispatch_GetPresentationDelay( IFusionSo
      
      ret = real->GetPresentationDelay( real, &delay );
      
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     ret, VOODOO_INSTANCE_NONE,
                                     VMBT_INT, delay,
                                     VMBT_NONE );
@@ -387,7 +387,7 @@ Dispatch_GetPlayback( IFusionSoundStream
           return ret;
      }
 
-     return voodoo_manager_respond( manager, msg->header.serial,
+     return voodoo_manager_respond( manager, true, msg->header.serial,
                                     DR_OK, instance,
                                     VMBT_NONE );
 }
--- FusionSound-1.1.1/tools/fsproxy.c.orig	2010-11-18 08:56:24.839107480 +0100
+++ FusionSound-1.1.1/tools/fsproxy.c	2010-11-18 09:30:10.919102452 +0100
@@ -162,7 +162,7 @@ server_run()
      DirectResult  ret;
      VoodooServer *server;
 
-     ret = voodoo_server_create( &server );
+     ret = voodoo_server_create( NULL, 0, false, &server );
      if (ret) {
           D_ERROR( "Voodoo/Proxy: Could not create the server (%s)!\n", FusionSoundErrorString(ret) );
           return ret;
--- FusionSound-1.1.1/src/core/core_sound.c.orig	2012-01-01 11:11:56.195907393 +0100
+++ FusionSound-1.1.1/src/core/core_sound.c	2012-01-01 11:57:20.872665292 +0100
@@ -314,7 +314,7 @@ fs_core_create_buffer( CoreSound *core )
      D_ASSERT( core->shared->buffer_pool != NULL );
 
      /* Create a new object in the buffer pool. */
-     return (CoreSoundBuffer*) fusion_object_create( core->shared->buffer_pool, core->world );
+     return (CoreSoundBuffer*) fusion_object_create( core->shared->buffer_pool, core->world, fusion_id(core->world) );
 }
 
 CorePlayback *
@@ -325,7 +325,7 @@ fs_core_create_playback( CoreSound *core
      D_ASSERT( core->shared->playback_pool != NULL );
 
      /* Create a new object in the playback pool. */
-     return (CorePlayback*) fusion_object_create( core->shared->playback_pool, core->world );
+     return (CorePlayback*) fusion_object_create( core->shared->playback_pool, core->world, fusion_id(core->world) );
 }
 
 DirectResult
