--- FusionSound-1.6.2/drivers/alsa.c.orig	2012-09-20 18:01:15.000000000 +0200
+++ FusionSound-1.6.2/drivers/alsa.c	2012-10-06 15:29:23.860469388 +0200
@@ -327,7 +327,7 @@
           if (snd_ctl_card_info( ctl, info ) == 0) {
                snprintf( device_info->name,
                          FS_SOUND_DEVICE_INFO_NAME_LENGTH,
-                         snd_ctl_card_info_get_name( info ) );
+                         "%s", snd_ctl_card_info_get_name( info ) );
           }
 
           snd_ctl_close( ctl );
